Index: condor-8.0.5~dfsg.1/src/condor_shadow.V6.1/CMakeLists.txt
===================================================================
--- condor-8.0.5~dfsg.1.orig/src/condor_shadow.V6.1/CMakeLists.txt	2013-12-29 06:24:37.000000000 -0500
+++ condor-8.0.5~dfsg.1/src/condor_shadow.V6.1/CMakeLists.txt	2014-01-14 15:53:03.408347075 -0500
@@ -26,3 +26,4 @@
 endif(NOT PROPER AND (LINUX OR DARWIN))
 condor_daemon( shadow "${shadowRmvElements}" "${CONDOR_LIBS_FOR_SHADOW}" "${C_SBIN}" OFF )
 
+TARGET_LINK_LIBRARIES(condor_shadow dl)
